client/src/components/sources/sourcesToolbar.js

```diff
@@ -0,0 +1,282 @@
+      filterValue
+    });
+  }
+
+  onValueKeyPress(keyEvent) {
+    const { filterValue, filterType } = this.props;
+
+    if (keyEvent.key === 'Enter' && filterValue && filterValue.length > 0) {
+      this.filterAdded(filterType, filterValue);
+      keyEvent.stopPropagation();
+      keyEvent.preventDefault();
+    }
+  }
+
+  removeFilter(filter) {
+    Store.dispatch({
+      type: dispatchTypes.REMOVE_FILTER,
+      filter
+    });
+  }
+
+  clearFilters() {
+    Store.dispatch({
+      type: dispatchTypes.CLEAR_FILTERS
+    });
+  }
+
+  updateCurrentSortType(sortType) {
+    Store.dispatch({
+      type: dispatchTypes.SET_SORT_TYPE,
+      sortType
+    });
+  }
+
+  toggleCurrentSortDirection() {
+    Store.dispatch({
+      type: dispatchTypes.TOGGLE_SORT_ASCENDING
+    });
+  }
+
+  renderFilterInput() {
+    const { filterType, filterValue } = this.props;
+    if (!filterType) {
+      return null;
+    }
+
+    if (filterType.filterType === 'select') {
+      return (
+        <Filter.ValueSelector
+          filterValues={filterType.filterValues}
+          currentValue={filterValue}
+          placeholder={filterType.placeholder}
+          onFilterValueSelected={this.filterValueSelected}
+        />
+      );
+    } else {
+      return (
+        <input
+          className="form-control"
+          type={filterType.filterType}
+          value={filterValue}
+          placeholder={filterType.placeholder}
+          onChange={e => this.updateCurrentValue(e)}
+          onKeyPress={e => this.onValueKeyPress(e)}
+        />
+      );
+    }
+  }
+
+  renderFilter() {
+    const { filterType } = this.props;
+
+    return (
+      <Filter>
+        <Filter.TypeSelector
+          filterTypes={SourceFilterFields}
+          currentFilterType={filterType}
+          onFilterTypeSelected={this.selectFilterType}
+        />
+        {this.renderFilterInput()}
+      </Filter>
+    );
+  }
+
+  renderSort() {
+    const { sortType, sortAscending } = this.props;
+
+    if (sortType) {
+      return (
+        <Sort>
+          <Sort.TypeSelector
+            sortTypes={SourceSortFields}
+            currentSortType={sortType}
+            onSortTypeSelected={this.updateCurrentSortType}
+          />
+          <Sort.DirectionSelector
+            isNumeric={sortType.isNumeric}
+            isAscending={sortAscending}
+            onClick={() => this.toggleCurrentSortDirection()}
+          />
+        </Sort>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  renderActions() {
+    return (
+      <div className="form-group">
+        <Button>Authenticate</Button>
+        <Button>Scan</Button>
+        <Button bsStyle="primary">Add</Button>
+      </div>
+    );
+  }
+
+  renderCounts() {
+    const { activeFilters, totalCount, filteredCount } = this.props;
+
+    return (
+      <h5>
+        {activeFilters && activeFilters.length > 0
+          ? filteredCount + ' of '
+          : null}
+        {totalCount + ' Result'}
```

Comment:
Optional: string literal might help with the pluralization... or not, having typed it out it's a little verbose
```js
{
`${totalCount} ${totalCount>1 ? 'Result': 'Results'}`
}
```